src/app/models/person.ts

```diff
@@ -47,19 +49,29 @@ export function getAuthorAffiliationsLabels(author: Person, affiliations: Affili
     .sort((a, b) => Number(a) - Number(b));
 }
 
-export function createAuthorsState(authorsXml: Element[]): Person[] {
-  return authorsXml.map((author) =>
-    createAuthor(author.getAttribute('id'), {
+export function createAuthorsState(authorsXml: Element[], notesXml: Element): Person[] {
+  const getCompetingInterestsXml = (authorEl: Element): Element => {
+    return Array.from(notesXml.querySelectorAll('[fn-type="COI-statement"]')).find((fnEl: Element) => {
+      const id = fnEl.getAttribute('id');
+      return authorEl.querySelector(`xref[ref-type="fn"][rid="${id}"]`);
+    });
+  };
+
+  return authorsXml.map((author) => {
+    const competingInterests = getCompetingInterestsXml(author);
+    return createAuthor(author.getAttribute('id'), {
       firstName: getTextContentFromPath(author, 'name > given-names'),
       lastName: getTextContentFromPath(author, 'name > surname'),
       suffix: getTextContentFromPath(author, 'name > suffix'),
       bio: createBioEditorState(author.querySelector('bio')),
       email: getTextContentFromPath(author, 'email'),
       orcId: getTextContentFromPath(author, 'contrib-id[contrib-id-type="orcid"]'),
       isCorrespondingAuthor: author.getAttribute('corresp') === 'yes',
-      affiliations: Array.from(author.querySelectorAll('xref[ref-type="aff"]')).map((xRef) => xRef.getAttribute('rid'))
-    })
-  );
+      affiliations: Array.from(author.querySelectorAll('xref[ref-type="aff"]')).map((xRef) => xRef.getAttribute('rid')),
+      hasCompetingInterest: Boolean(competingInterests),
```

Comment:
This needs to be more clever, as in the input XML there technically is a competing interest for each author although in the majority of cases that value is always 'No competing interests declared', as such I think something like...
```
hasCompetingInterest(text) {
    return text && text !== "No competing interests declared";
}
```

Reply:
Yeah I thought it looked rather dumb but assumed the xml isn't right.

Reply:
I makes sense from the XML perspective, but yeah it's quirky to handle when loading the content.